### PR TITLE
Improve FindEOSPAC.cmake

### PR DESCRIPTION
**Pull request by KineticTheory**
### Background

* eospac is now published by spack, so we need to upgrade the `FindEOSPAC.cmake` module to find eospac installed in a _normal_ way.  Previously, we relied on hand rolled environment variables to help the build system find `libeospac.a`.

### Description of changes

+ Teach `FindEOSPAC.cmake` to find eospac that is provided via spack and without environment variables like `EOSPAC_LIB_DIR`.
+ Update the `FindEOSPAC.cmake` script to use modern CMake recommendations, including a populated link target `EOSPAC::eospac`.
+ Move eospac discovery from `cdi_eospac/CMakeLists.txt` to `config/vendor_libraries.cmake`.
+ In `cdi_eospac/CMakeLists.txt` use the new `EOSPAC::eospac` build target as a link dependency instead of cmake variables like `${EOSPAC_LIBRARY}`.

### Status

* Reference:
  * [Pre-Merge Code Review](https://github.com/lanl/Draco/wiki/Style-Guide)
  * [CDash Status](https://rtt.lanl.gov/cdash/index.php?project=Draco&filtercount=1&showfilters=1&field1=buildname&compare1=63&value1=-pr)
* Checks
  * [x] Travis CI checks pass
  * [x] Code coverage does not decrease
  * [x] [Valgrind test passes](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] [Toss3 checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] [Trinitite checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [ ] Code reviewed/approved, sufficient DbC checks, testing, documentation

### CODE REVIEW

**Comment by KineticTheory** [comment]
Marking as WIP - I accidentally included changes from #388.  I'll need to fix the changeset or let #388 be merged first.

File: config/FindEOSPAC.cmake [hunk 1]
```diff
@@ -5,81 +5,201 @@
...
 find_library(EOSPAC_LIBRARY
-    NAMES ${EOSPAC_LIBRARY_NAME}
-    PATHS
-        ${EOSPAC_LIB_DIR}
-        $ENV{EOSPAC_LIB_DIR}
-        $ENV{VENDOR_DIR}/lib
-        ${VENDOR_DIR}/lib
-    NO_DEFAULT_PATH
+  NAMES ${EOSPAC_LIBRARY_NAME}
+  PATHS ${COMPTON_ROOT_DIR}/lib
```
Comment (RyanWollaeger) [inline config/FindEOSPAC.cmake]: Is `${COMPTON_ROOT_DIR}` the correct path for this library?

File: config/FindEOSPAC.cmake [hunk 2]
```diff
@@ -5,81 +5,201 @@
...
-EOSPAC_INCLUDE_DIRS = ${EOSPAC_INCLUDE_DIR}
-EOSPAC_LIBRARIES    = ${EOSPAC_LIBRARY}
-")
+if( EOSPAC_FOUND AND NOT TARGET EOSPAC::eospac )
+  if( EXISTS "${EOSPAC_LIBRARY_DLL}" )
+
+    # Windows systems with dll libraries.
+    add_library( EOSPAC::eospac SHARED IMPORTED )
+
+    # Windows with dlls, but only Release libraries.
```
Comment (RyanWollaeger) [inline config/FindEOSPAC.cmake]: In the `find_library` command for `EOSPAC_LIBRARY` above, the path suffixes include `Debug`.  If a release version was not found, would the following `set_target_properties` exclude the Debug library?

Comment (KineticTheory) [reply config/FindEOSPAC.cmake]: For Makefile builds, there will only be one library.  The Debug/Release subfolders are typically used by multi-config build tools like Visual Studio or Xcode.  These build tools know what to do with the Debug/Release information.

**Pull request merged**